### PR TITLE
Issue161: profile page

### PR DESCRIPTION
**Pull request by esteeroramy**
image
image
image
NOTE: the quizzard logo is there temporarily until the file system is implemented

### CODE REVIEW

File: public/js/profile.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+* Submits the new updates for the profile
+*/
+$('#editForm').submit(function(evt) {
+    evt.preventDefault();
+
+    if ($('#passwd').val() === $('#confirmpasswd').val()) {
+        var id = $('#userId').html();
```
Comment (ahmedqarmout2) [inline public/js/profile.js:8]: We need to check that both password and confirm password are the same in the backend side as well as the api

File: public/js/profile.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* @method disableEdit
+*/
+var disableEdit = function() {
+    location.reload();
+}
+
+var editProfile = function(id) {
+    var fields = $('#editForm').serializeArray();
+    var user = {
+        originalID: id
```
Comment (ahmedqarmout2) [inline public/js/profile.js]: This is not needed anymore, please remove it. Also we can for all form fields down here, don't we just need a few of them.

File: public/js/profile.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+        data: user,
+        success: function(data) {
+            location.reload();
+        },
+        error: function(data) {
+            if (data['status'] === 401) {
+                window.location.href = '/';
+            } else if (data['status'] === 403) {
+                failSnackbar('Current Password is Incorrect');
+            } else if (data.result === 'failure') {
```
Comment (ahmedqarmout2) [inline public/js/profile.js]: This if and the one below it are not needed as we don't send these values back from the backend. Plz remove. So just keep the last else

File: server/db.js [hunk 4]
```diff
@@ -315,32 +315,41 @@ var updateUserById = function(userId, info, callback){
             return callback(null, 'success');
         });
     } else {
-        bcrypt.hash(info.newPassword, 11, function(err, hash) {
+      update.$set.password = info.newPassword;
```
Comment (ahmedqarmout2) [inline server/db.js]: Indentation is off

**Review by ahmedqarmout2 — CHANGES_REQUESTED** [review]
Nice work fam. Looks good overall, just a few comments

File: public/js/profile.js (hunk 1 above)
Comment (mehtadarshan1) [reply public/js/profile.js]: agree

Comment (mehtadarshan1) [reply public/js/profile.js:8]: Yes, better to have backend validation as well, since front end can fail

**Review by ahmedqarmout2 — APPROVED** [review]

**Review by mehtadarshan1 — APPROVED** [review]
looks good

**Review by Mohammed-JS-Khan — APPROVED** [review]

**Pull request merged**